src/request/derive-address/DeriveAddressApi.js

```diff
@@ -0,0 +1,61 @@
+/* global TopLevelApi */
+/* global DeriveAddress */
+/* global Key */
+/* global KeyStore */
+/* global Nimiq */
+
+class DeriveAddressApi extends TopLevelApi { // eslint-disable-line no-unused-vars
+    /**
+     * @param {DeriveAddressRequest} request
+     */
+    async onRequest(request) {
+        const parsedRequest = await DeriveAddressApi._parseRequest(request);
+        const handler = new DeriveAddress(parsedRequest, this.resolve.bind(this), this.reject.bind(this));
+        handler.run();
+    }
+
+    /**
+     * @param {DeriveAddressRequest} request
+     * @returns {Promise<ParsedDeriveAddressRequest>}
+     * @private
+     */
+    static async _parseRequest(request) {
+        if (!request) {
+            throw new Error('Empty request');
+        }
+
+        if (!request.appName || typeof request.appName !== 'string') {
+            throw new Error('appName is required');
+        }
+
+        if (!request.keyId || typeof request.keyId !== 'string') {
+            throw new Error('keyId is required');
+        }
+
+        // Check that key exists.
+        const keyInfo = await KeyStore.instance.getInfo(request.keyId);
+        if (!keyInfo) {
+            throw new Error('Unknown keyId');
+        }
+
+        if (keyInfo.type === Key.Type.LEGACY) {
+            throw new Error('Cannot derive addresses for single-account wallets');
+        }
+
+        if (!request.baseKeyPath || !Nimiq.ExtendedPrivateKey.isValidPath(request.baseKeyPath)) {
+            throw new Error('Invalid defaultKeyPath');
+        }
+
+        if (!request.indicesToDerive || !(request.indicesToDerive instanceof Array)) {
+            throw new Error('Invalid indicesToDerive');
```

Comment:
Needs to be an array of strings, actually, but I believe numbers would work as well, because the derivation code in `Nimiq.ExtendedPrivateKey` automatically handles non-hardened indices.